src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -53,7 +59,18 @@ public AddCommand parse(String args) throws ParseException {
         Optional<Preference> preference = preferenceArg.isPresent()
                 ? Optional.of(ParserUtil.parsePreference(preferenceArg.get()))
                 : Optional.empty();
-        UserType userType = ParserUtil.parseUserType(argMultimap.getValue(PREFIX_USERTYPE).get());
+        // if the user did not input any properties or preference, throw an error as user needs to be a buyer or seller
+        if (properties.isEmpty() && preferenceArg.isEmpty() || !properties.isEmpty() && preferenceArg.isPresent()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, UserType.MESSAGE_USAGE));
+        }
```

Comment:
Is this check needed since you already did this

```java
        if (argMultimap.getValue(PREFIX_PREFERENCE).isPresent() && argMultimap.getValue(PREFIX_PROPERTY).isPresent()
            || argMultimap.getValue(PREFIX_PREFERENCE).isEmpty() && argMultimap.getValue(PREFIX_PROPERTY).isEmpty()) {
            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, UserType.MESSAGE_USAGE));
        }
```